docs/en/03_Upgrading/04_Upgrading_graphql4.md

```diff
@@ -0,0 +1,117 @@
+
+`silverstripe/graphql` v4 is a complete rewrite and provides substantial performance improvements.
+
+`silverstripe/graphql` v4 provides developers a first class tool for building APIs and allowing third party services to integrate with their Silverstripe CMS websites.
+
+## That sounds risky, do I absolutely have to use version 4?
+
+Silverstripe CMS has been shipping with dual support for `silverstripe/graphql` v3 and v4 since the 4.8 release. Until now `silverstripe/graphql` v4 had been in alpha and you had to explicitly opt-in to get it. At Silverstripe, we are already using `silverstripe/graphql` v4 in production on several projects.
+
+All the supported Silverstripe CMS modules that use `silverstripe/graphql` have dual-support for version 3 and version 4. If you wish to stay on `silverstripe/graphql` v3, you can do so and it will not block you from upgrading to Silverstripe CMS 4.11.
+
+We will maintain support for `silverstripe/graphql` v3 in Silverstripe CMS 4 for the foreseeable future. Any change to this policy will be announced at least 6 months in advance. 
+
+### Opting out of `silverstripe/graphql` v4 and sticking to version 3
+
+If your project composer.json file already explicitly requires silverstripe/graphql, you don’t need to do anything.
+
+If your project uses silverstripe/recipe-cms, composer will try to install `silverstripe/graphq` v4.0 when you upgrade to the 4.11 release. To stay on silverstripe/graphql:^3, you’ll need to explicitly require `silverstripe/graphql` v3.8.
+
+```
+composer require silverstripe/graphql:^3
+```
+
+To validate which version of `silverstripe/graphql` your project is using, run this composer command:
+`composer show silverstripe/graphql`
+
+To view which dependencies require `silverstripe/graphql`, run this composer command:
+`composer why silverstripe/graphql`
+
+## How do I get this thing working?
+
+Part of the reason why `silverstripe/graphql` v4 is so much faster than v3 is that it has a “code generation” step. Silverstripe CMS will generate PHP classes for your GraphQL schemas and stores them in a `.graphql-generated` folder in the root of your project.
+
+### What triggers a GraphQL code build? 
+
+- This folder will automatically be generated when you run a `dev/build` on your project. 
+- You can also run the `dev/graphql/build` command to explicitly build your GraphQL schemas. 
+- Silverstripe CMS will attempt to generate your schema on the first graphql request if it wasn’t already generated.
+
+### Deploying your `.graphql-generated` folder 
+
+One way or another, you must get this `.graphql-generated` folder into your production environment for Silverstripe CMS to work as expected. There are many ways to do so.
+
+#### Commit `.graphql-generated`
```

Comment:
My instinct is to go from the simplest option to the most complicated one. I thought this one was the simplest.

@emteknetnz thought having this option first it implied this was the recommended way of doing things.

@GuySartorelli Cut this baby in half please.

Reply:
I think the "however it will pollute your commits with massive diff for the generated code" line indicates pretty clearly that this isn't the preferred approach - but if there is doubt, maybe update that line to explicitly say "this is not the preferred approach because ..."

Reply:
This is moved to `05_deploying_the_schema.md` which explicitly states that the options are listed in order of complexity.